pkg/apis/core/validation/shoot.go

```diff
@@ -1291,7 +1291,11 @@ func ValidateKubeAPIServer(kubeAPIServer *core.KubeAPIServerConfig, version stri
 		return allErrs
 	}
 
-	if oidc := kubeAPIServer.OIDCConfig; oidc != nil {
+	// TODO(AleksandarSavchev): Remove this check as soon as v1.32 is the least supported Kubernetes version in Gardener.
+	k8sGreaterEqual132, _ := versionutils.CheckVersionMeetsConstraint(version, ">= 1.32")
+	if oidc := kubeAPIServer.OIDCConfig; k8sGreaterEqual132 && oidc != nil {
```

Comment:
Also add a TODO statement (or even implementation [here](https://github.com/gardener/gardener/blob/a37341db3a878d05434170a51c498f23807b7299/pkg/controllermanager/controller/shoot/maintenance/reconciler.go#L177) in the maintenance controller for forceful Kubernetes version upgrades).